src/functions-and-arrays.js

```diff
@@ -1,38 +1,109 @@
 // Iteration #1: Find the maximum
-function maxOfTwoNumbers() {}
+function maxOfTwoNumbers(num1, num2) {
+  if (num1>num2){
+    return num1;
+  } else {
+    return num2;
+  }
+}
 
 
 
 // Iteration #2: Find longest word
 const words = ['mystery', 'brother', 'aviator', 'crocodile', 'pearl', 'orchard', 'crackpot'];
 
-function findLongestWord() {}
+function findLongestWord(words) {
+  if (words.length===0){
+    return null;
+  }
+ let longestWord = words[0]
+
+  for (let i = 0; i < words.length; i++) {
+    const currentWord = words[i]
+    if (currentWord.length > longestWord.length) {
+      longestWord = currentWord
+    }
+  }
+  return longestWord
 
+}
+const longestWord = findLongestWord(words);
 
 
 // Iteration #3: Calculate the sum
 const numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
 
-function sumNumbers() {}
+function sumNumbers(numbers) {
+  let sum = 0;
+  for (let i = 0; i < numbers.length; i++) {
+    sum += numbers[i];
+  }
+  return sum;
+}
 
 
 
 // Iteration #3.1 Bonus:
-function sum() {}
-
+function sum(arr) {
+  let totalSum = 0;
+
+  for (let i = 0; i < arr.length; i++) {
+    const element = arr[i];
+
+    if (typeof element === 'number' || !isNaN(parseFloat(element))) {
+      totalSum += Number(element);
+    } else if (typeof element === 'string') {
+      totalSum += element.length;
+    } else if (typeof element === 'boolean') {
+      totalSum += element ? 1 : 0;
+    }
+  }
+
+  return totalSum;
+}
 
+const mixedArr = [6, 12, 'miami', 1, true, 'barca', '200', 'lisboa', 8, 10];
 
 // Iteration #4: Calculate the average
 // Level 1: Array of numbers
+function averageNumbers(numbers) {
+  if (numbers.length === 0) {
+    return null;
+  }
+
+  let sum = 0;
+  for (let i = 0; i < numbers.length; i++) {
+    sum += numbers[i];
+  }
+  const average = sum / numbers.length;
```

Comment:
Esta oarte de codigo yo la veo ya en tu codigo
Mira te lo hago una vez, si te fijas esto ya esta en sum, es decir que podriamos ejecutar, digo bien, ejecutar la funcion sum, por ello:
```suggestion
  const average = sum(numbers) / numbers.length;
```

Y asi te ahorras escribir codigo de mas y la abstraccion queda mucho mejor!